// @strapi/admin/dist/admin/admin/src/services/homepage.js
'use strict';

var api = require('./api.js');

const homepageService = api.adminApi.enhanceEndpoints({
    addTagTypes: [
        'CountDocuments'
    ]
}).injectEndpoints({
    endpoints: (builder)=>({
            getKeyStatistics: builder.query({
                query: ()=>'/admin/homepage/key-statistics',
                transformResponse: (response)=>response.data,
                providesTags: (_, _err)=>[
                        'HomepageKeyStatistics'
                    ]
            }),
            getCountDocuments: builder.query({
                query: ()=>'/content-manager/homepage/count-documents',
                transformResponse: (response)=>response.data,
                providesTags: (_, _err)=>[
                        'CountDocuments'
                    ]
            })
        })
});
const { useGetKeyStatisticsQuery, useGetCountDocumentsQuery } = homepageService;

exports.useGetCountDocumentsQuery = useGetCountDocumentsQuery;
exports.useGetKeyStatisticsQuery = useGetKeyStatisticsQuery;
